python/README.md

```diff
@@ -38,6 +38,28 @@ Testing is done using tox. The config can be found in `tox.ini` within the pytho
 tox
 ```
 
+#### `s3` Tests
+
+A subset of tests are decorated using `@pytest.mark.s3` and require an S3 protocol compliant object storage to use for testing. For the python CI, a local MinIO container is used. To run these tests, you can provide a pass-through argument to tox (using `--`) that will be passed to the `pytest` command.
+
+```
+tox -- -m "s3"
+```
+
+Additional arguments can be passed to configure the object storage.
+
+| Argument                | Default               | Description                                         |
+|-------------------------|-----------------------|-----------------------------------------------------|
+| --endpoint-url          | http://localhost:9000 | The S3 endpoint URL for tests marked as s3          |
+| --aws-access-key-id     | admin                 | The AWS access key ID for tests marked as s3        |
+| --aws-secret-access-key | password              | The AWS secret access key ID for tests marked as s3 |
+
+For example, to run `s3` tests against an object store running locally on port 4000 with an access key ID of `foo` and an secret access key as `bar`, you would use the following command.
+
+```
+tox -- -m "s3" --endpoint-url=http://localhost:4000 --aws-access-key-id=foo --aws-secret-access-key=bar
```

Comment:
Using this, you could run a minio instance locally (or use an aws s3 bucket) and run the tests marked as `s3` against that.

Reply:
I think this is going to be out of date now that we're using pre-commit.